modules/core/src/hal_replacement.hpp

```diff
@@ -472,8 +472,93 @@ inline int hal_ni_dctFree2D(cvhalDFT *context) { return CV_HAL_ERROR_NOT_IMPLEME
 #define cv_hal_dctFree2D hal_ni_dctFree2D
 //! @endcond
 
+
+//! @addtogroup core_hal_interface_matrix_decomposition Matrix decomposition
+//! @{
+
+/**
```

Comment:
I think it will be better to use three separate documentation groups for each decomposition type, then we will be able to document each family separately:
```.cpp
/**
<desc>
@param ...
@param ...
*/
//! @addtogroup core_hal_interface_decomp_lu LU matrix decomposition
//! @{
...
//! @}

/**
<desc>
@param ...
@param ...
*/
//! @addtogroup core_hal_interface_decomp_cholesky Cholesky matrix decomposition
//! @{
...
//! @}

...
```
With this method the documentation will look like this: http://docs.opencv.org/master/df/d2e/group__core__hal__interface__merge.html
The group description contains function and parameters description and each function itself is not documented.